app/assets/javascripts/discourse/app/components/d-styles.gjs

```diff
@@ -0,0 +1,90 @@
+import Component from "@glimmer/component";
+import { service } from "@ember/service";
+import { getURLWithCDN } from "discourse-common/lib/get-url";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class DStyles extends Component {
+  @service session;
+  @service site;
+
+  @bind
+  categoryColors() {
```

Comment:
I think you probably do want getters here rather than bound functions. 

Why did you switch to functions? Is it because changes to `site.categories` weren't being updated live? If so, then I think the solution is to add `@tracked categories` in the `Site` model.

(or to avoid `@tracked`, do `this.site.get("categories")` rather than `this.site.categories`. That's the way to achieve inter-compatibility between the legacy EmberObject system and modern autotracking)